### PR TITLE
Add Eco-Metabolomics

### PR DESCRIPTION
**Pull request by korseby**
Hi all,

please add the Eco-Metabolomics section to Galaxy.

PR is against develop.

The PR includes all tools and the workflow. The tools work out of the box, but import modules need to be adjusted to work with the latest ISA-tools.

This is the initial PR. Please list remaining issues.

Best wishes, *Kristian

### CODE REVIEW

**Comment by ilveroluca** [comment]
Hi @korseby. The new tools load perfectly.  There are some things that I don't quite understand though.

* Why integrate another MTBLS dataset downloader, when we already have a tool for that?
* All the new tools make reference to a specific dataset (mtbls520).  Could I use them to apply the same analysis on a different dataset?

**Comment by ilveroluca** [comment]
In (this example)[https://github.com/phnmnl/container-galaxy-k8s-runtime/compare/ecomet1.1...ilveroluca:ecomet1.1] I've generalized (I think!) one of the tools, taking out all references to MTBLS520.

Note that if you want to share the specific analysis you performed for the MTBLS520 study, we can assemble a workflow that uses the generic tools but fixes the parameters, which (given the downloader) would also fix the input dataset and thus reproduce the same output result.

**Comment by korseby** [comment]
Hi @ilveroluca ,

I have updated the naming conventions. I would like to keep the separate mtbls520 importers and extractors until the ISA-Download modules works.

Can you have a look again?

Best wishes, *Kristian

**Comment by ilveroluca** [comment]
What about these four tools?

- mtbls520_02a_raw_extract
- mtbls520_02b_qc_extract
- mtbls520_02_extract
- mtbls520_01_mtbls_download

**Comment by korseby** [comment]
These tools import the mtbls dataset. I want to keep them until the ISA-Importer work. Currently, I have some problems creating the dataset collections that contain  only the study-mzML and only the QC-mzML in pos/neg-mode. Same thing for the NMR workflow by the way.

**Comment by djcomlab** [comment]
@sneumann I've started work on `ISAslicer2` to build in the more fully featured querying - query by Factor Values, Material Type and assay type (Measurement/Technology types).

Splitting by QC/study I guess would be on `Material Type` or similar?

**Comment by ilveroluca** [comment]
@korseby can we make them generic, so that the user can insert the dataset id as a parameter? (i.e., suppose I wanted to try them on an MTBLS dataset different from MTBLS520)

**Comment by korseby** [comment]
The preferred method is to use ISA-Slicer and/or one of the other ISA tools. However, before I can test the workflow I need the customized MTBLS520 tools - because there are no other tools that provide that functionality and no other way to create the required dataset collections (because files are too large and manual upload does not work except ftp which is broken). The files are required for the entire workflow and currently I can not replace the above tools with the ISA tools.

You can give a different MTBLS ID than 520 to `mtbls520_01_mtbls_download`, the tool would download the entire study with a different ID. The `mtbls520_02*` are rather specific and extract the study files in pos or neg mode (specified as a parameter) and/or the QC files in pos or neg mode (specified as a parameter and by using a different tool). The entire workflow is a rather specific use case and it will not be possible before the end of phnmnl to make the workflow more generic, simply because there are no (published) "Eco-Metabolomics" studies on Metabolights yet.

**Comment by korseby** [comment]
@djcomlab There are two assays: pos and neg mode. The easiest way would be to parse all files referenced in 'Raw Spectral Data File' and take all files except those that include "blank value" in 'Factor Value[species]' or vice versa. For the workflow, I also need the associated assay file, study file and maf files in the correct polarity.

**Comment by korseby** [comment]
I updated to version tag of the container for the stable release.

**Comment by korseby** [comment]
Hi @ilveroluca , I made the workflow and tools more generic. Please review.

**Comment by ilveroluca** [comment]
As it stands, it looks like the tools need to be updated to reflect the generalization implemented in these wrappers.  For instance, the downloader fails if you try to download anything other than MTBLS520.  Also, the tool container should follow our [Dockerfile guidelines](https://github.com/phnmnl/phenomenal-h2020/wiki/Dockerfile-Guide#reduce-image-size-by-combining-multiple-line-runs)

**Comment by korseby** [comment]
Updated as requested.

File: tools/phenomenal/ecomet/ecomet_11_species_nmds.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    <param name="infile1" type="data" format="rdata" optional="False" label="Input Rdata file" />
+  </inputs>
+  <outputs>
+    <data name="outfile1" type="data" format="pdf" label="Species NMDS Stress" />
+    <data name="outfile2" type="data" format="pdf" label="Species NMDS Procrustes" />
+    <data name="outfile3" type="data" format="pdf" label="Species NMDS" />
+  </outputs>
+  <help><![CDATA[
+This is the MTBLS520 study.
+The study will be published soon. This code is work-in-progress. Do not redistribute unless study published.
```
Comment (ilveroluca) [inline tools/phenomenal/ecomet/ecomet_11_species_nmds.xml]: @korseby:  given the message here, are you sure you want to include this tool in PhenoMeNal at this time?

File: tools/phenomenal/ecomet/ecomet_24_seasons_nmds.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    <param name="infile1" type="data" format="rdata" optional="False" label="Input Rdata file" />
+  </inputs>
+  <outputs>
+    <data name="outfile1" type="data" format="pdf" label="Goodness of Fit and Shepard Plot for Seasons NMDS" />
+    <data name="outfile2" type="data" format="pdf" label="Procrustes analysis for original and fitted Seasons NDMS" />
+    <data name="outfile3" type="data" format="pdf" label="Seasons NMDS plot" />
+  </outputs>
+  <help><![CDATA[
+This is the MTBLS520 study.
+The study will be published soon. This code is work-in-progress. Do not redistribute unless study published.
```
Comment (ilveroluca) [inline tools/phenomenal/ecomet/ecomet_24_seasons_nmds.xml]: Same goes for this tool :-)

**Pull request merged**